typescript/demo/function-call-stream.ts

```diff
@@ -217,12 +222,20 @@ async function functionCallingWithAIConfig() {
       return;
     }
 
-    const rawResponse = output.metadata?.rawResponse;
-    const functionCall = rawResponse?.function_call;
+    let functionCall: FunctionData | null = null;
+    if (output.data?.hasOwnProperty("value")) {
+      const outputData = output.data as OutputDataWithValue;
+      if (outputData.kind === "tool_calls") {
+        outputData as OutputDataWithToolCallsValue;
+        // Typescript schema does not support array of function calls yet,
+        // but Python does so doing this for forward compatibility
+        functionCall = outputData.value[outputData.value.length - 1].function;
+      }
+    }
     console.log("functionCall=", functionCall);
 
     // If there is no function call, we're done and can exit this loop
-    if (!functionCall) {
+    if (functionCall === null) {
```

Comment:
This is incorrect. 

`if (!functionCall)` checks for both `null` or `undefined`, whereas `if (functionCall === null)` only checks for `null`.

Reply:
I initialized it to null so it can never be undefined in this flow, but yea I get what you're saying I can check for both to ensure it's future-proofed